Src/Base/AMReX_ParmParse.cpp

```diff
@@ -867,7 +867,7 @@ squeryarr (const ParmParse::Table& table,
     if ( num_val == 0 ) return true;
 
     int stop_ix = start_ix + num_val - 1;
-    if ( static_cast<int>(ref.size()) <= stop_ix )
+    if ( static_cast<int>(ref.size()) != stop_ix + 1 )  // grow or shrink to fit
```

Comment:
I am not yet 100% happy with the fix, since it might resize a vector where the API might allow to only partially read values from the params....?

Not 100% sure about this API contract, as the changes on the passed vector are only partly documented in the header file.

If needed, we could also apply a resize fix one level up in `queryAdd` for `std::vector`...?

Reply:
The doxygen string (for `getktharr`) does say "ref is resized (if necessary) to hold num_val_values".  The doxygen string for `queryarr` says "Same as ...".

On one hand, it seems a good idea to resize the vector so that the size of the vector tells us how many values ParmParse has.  Otherwise, we have to call `countval` to find it out. On the other hand, it's possible it might break some people's code. So I don't know.

It seems that the doxygen string for `getktharr` is wrong on what `start_ix` is.  It says
```
The value in the list indexed by start_ix is copied into std::vector<int>[0],
std::vector<int>[1] holds start_ix+1, etc.
```
But it seems to me the code puts the value in the list indexed by `start_ix` into `std::vector<int>[start_ix]`, not `[0]`.

Anyway, I don't think I ever used the two optional parameters, `start_ix` and `num_val`.

Reply:
If we do change it to always resize, we don't need to do `if ( static_cast<int>(ref.size()) != stop_ix + 1 )  // grow or shrink to fit`.  Just `resize` will work for all cases.

Reply:
If we have some codes that actually rely on the no resize unless necessary behavior, @asalmgren and @drummerdoc might know.

Reply:
Let me push the same high-level fix as I used in ImpactX as a work-around.
It applies only to one overload of `queryAdd` where it is definitely safe to resize.